control/controlhttp/client.go

```diff
@@ -65,7 +64,7 @@ const (
 //
 // The provided ctx is only used for the initial connection, until
 // Dial returns. It does not affect the connection once established.
-func Dial(ctx context.Context, addr string, machineKey key.MachinePrivate, controlKey key.MachinePublic, protocolVersion uint16) (*controlbase.Conn, error) {
+func Dial(ctx context.Context, addr string, machineKey key.MachinePrivate, controlKey key.MachinePublic, protocolVersion uint16, dialer *tsdial.Dialer) (*controlbase.Conn, error) {
```

Comment:
does this actually need the whole `*tsdial.Dialer` type or just the dial func?

Reply:
Dial func is sufficient, done